diff --git a/TestVectors/runtimes/java/src/main/smithy-generated/software/amazon/cryptography/encryptionsdk/wrapped/TestESDK.java b/TestVectors/runtimes/java/src/main/smithy-generated/software/amazon/cryptography/encryptionsdk/wrapped/TestESDK.java
index 1dc9dd24..262bb622 100644
--- a/TestVectors/runtimes/java/src/main/smithy-generated/software/amazon/cryptography/encryptionsdk/wrapped/TestESDK.java
+++ b/TestVectors/runtimes/java/src/main/smithy-generated/software/amazon/cryptography/encryptionsdk/wrapped/TestESDK.java
@@ -4,9 +4,16 @@
 package software.amazon.cryptography.encryptionsdk.wrapped;
 
 import Wrappers_Compile.Result;
+import com.amazonaws.encryptionsdk.CryptoAlgorithm;
+import com.amazonaws.encryptionsdk.CryptoResult;
+import dafny.DafnyMap;
+import dafny.DafnySequence;
 import java.lang.IllegalArgumentException;
 import java.lang.RuntimeException;
+import java.nio.ByteBuffer;
+import java.util.Map;
 import java.util.Objects;
+import com.amazonaws.encryptionsdk.AwsCrypto;
 import software.amazon.cryptography.encryptionsdk.ESDK;
 import software.amazon.cryptography.encryptionsdk.ToDafny;
 import software.amazon.cryptography.encryptionsdk.ToNative;
@@ -16,10 +23,12 @@ import software.amazon.cryptography.encryptionsdk.internaldafny.types.EncryptInp
 import software.amazon.cryptography.encryptionsdk.internaldafny.types.EncryptOutput;
 import software.amazon.cryptography.encryptionsdk.internaldafny.types.Error;
 import software.amazon.cryptography.encryptionsdk.internaldafny.types.IAwsEncryptionSdkClient;
+import software.amazon.cryptography.materialproviders.internaldafny.types.ESDKAlgorithmSuiteId;
+import software.amazon.smithy.dafny.conversion.ToDafny.Simple;
 
 public class TestESDK implements IAwsEncryptionSdkClient {
 
-  private final ESDK _impl;
+  private final AwsCrypto _impl;
 
   protected TestESDK(BuilderImpl builder) {
     this._impl = builder.impl();
@@ -33,9 +42,30 @@ public class TestESDK implements IAwsEncryptionSdkClient {
     try {
       software.amazon.cryptography.encryptionsdk.model.DecryptInput nativeInput =
         ToNative.DecryptInput(dafnyInput);
-      software.amazon.cryptography.encryptionsdk.model.DecryptOutput nativeOutput =
-        this._impl.Decrypt(nativeInput);
-      DecryptOutput dafnyOutput = ToDafny.DecryptOutput(nativeOutput);
+      final CryptoResult<byte[], ?> decryptResult;
+
+      if (Objects.isNull(nativeInput.materialsManager())) {
+        // Call decrypt with keyring
+        if (Objects.isNull(nativeInput.encryptionContext())) {
+          decryptResult = this._impl.decryptData(nativeInput.keyring(), nativeInput.ciphertext().array());
+        } else {
+          decryptResult = this._impl.decryptData(nativeInput.keyring(), nativeInput.ciphertext().array(), nativeInput.encryptionContext());
+        }
+      } else {
+        if (Objects.isNull(nativeInput.encryptionContext())) {
+          decryptResult = this._impl.decryptData(nativeInput.materialsManager(), nativeInput.ciphertext().array());
+        } else {
+          decryptResult = this._impl.decryptData(nativeInput.materialsManager(), nativeInput.ciphertext().array(), nativeInput.encryptionContext());
+        }
+      }
+      DafnySequence<? extends Byte> plaintext = Simple.ByteSequence(decryptResult.getResult());
+      DafnyMap<? extends DafnySequence<? extends Byte>, ? extends DafnySequence<? extends Byte>> encryptionContext =
+        software.amazon.cryptography.materialproviders.ToDafny.EncryptionContext(decryptResult.getEncryptionContext());
+      ESDKAlgorithmSuiteId algorithmSuiteId = software.amazon.cryptography.materialproviders.ToDafny.ESDKAlgorithmSuiteId(
+        decryptResult.getCryptoAlgorithm().getAlgorithmSuiteId().ESDK()
+      );
+      DecryptOutput dafnyOutput = new DecryptOutput(plaintext, encryptionContext, algorithmSuiteId);
+
       return Result.create_Success(
         DecryptOutput._typeDescriptor(),
         Error._typeDescriptor(),
@@ -54,9 +84,35 @@ public class TestESDK implements IAwsEncryptionSdkClient {
     try {
       software.amazon.cryptography.encryptionsdk.model.EncryptInput nativeInput =
         ToNative.EncryptInput(dafnyInput);
-      software.amazon.cryptography.encryptionsdk.model.EncryptOutput nativeOutput =
-        this._impl.Encrypt(nativeInput);
-      EncryptOutput dafnyOutput = ToDafny.EncryptOutput(nativeOutput);
+      final CryptoResult<byte[], ?> encryptResult;
+
+      // Java ESDK is special and you have to set the algorithm suite both in the keyring which the
+      // test vectors do, but also in the client itself.
+      CryptoAlgorithm cryptoAlgorithm = _getAlgorithmSuite(nativeInput.algorithmSuiteId());
+      this._impl.setEncryptionAlgorithm(cryptoAlgorithm);
+
+      if (Objects.isNull(nativeInput.materialsManager())) {
+        // Call decrypt with keyring
+        if (Objects.isNull(nativeInput.encryptionContext())) {
+          encryptResult = this._impl.encryptData(nativeInput.keyring(), nativeInput.plaintext().array());
+        } else {
+          encryptResult = this._impl.encryptData(nativeInput.keyring(), nativeInput.plaintext().array(), nativeInput.encryptionContext());
+        }
+      } else {
+        if (Objects.isNull(nativeInput.encryptionContext())) {
+          encryptResult = this._impl.encryptData(nativeInput.materialsManager(), nativeInput.plaintext().array());
+        } else {
+          encryptResult = this._impl.encryptData(nativeInput.materialsManager(), nativeInput.plaintext().array(), nativeInput.encryptionContext());
+        }
+      }
+      dafny.DafnySequence<? extends Byte> ciphertext = Simple.ByteSequence(encryptResult.getResult());
+      DafnyMap<? extends DafnySequence<? extends Byte>, ? extends DafnySequence<? extends Byte>> encryptionContext =
+        software.amazon.cryptography.materialproviders.ToDafny.EncryptionContext(encryptResult.getEncryptionContext());
+      ESDKAlgorithmSuiteId algorithmSuiteId = software.amazon.cryptography.materialproviders.ToDafny.ESDKAlgorithmSuiteId(
+        encryptResult.getCryptoAlgorithm().getAlgorithmSuiteId().ESDK()
+      );
+
+      EncryptOutput dafnyOutput = new EncryptOutput(ciphertext, encryptionContext, algorithmSuiteId);
       return Result.create_Success(
         EncryptOutput._typeDescriptor(),
         Error._typeDescriptor(),
@@ -71,26 +127,55 @@ public class TestESDK implements IAwsEncryptionSdkClient {
     }
   }
 
+  private CryptoAlgorithm _getAlgorithmSuite(software.amazon.cryptography.materialproviders.model.ESDKAlgorithmSuiteId esdkAlgorithmSuiteId) {
+    switch (esdkAlgorithmSuiteId) {
+      case ALG_AES_128_GCM_IV12_TAG16_NO_KDF:
+        return CryptoAlgorithm.ALG_AES_128_GCM_IV12_TAG16_NO_KDF;
+      case ALG_AES_192_GCM_IV12_TAG16_NO_KDF:
+        return CryptoAlgorithm.ALG_AES_192_GCM_IV12_TAG16_NO_KDF;
+      case ALG_AES_256_GCM_IV12_TAG16_NO_KDF:
+        return CryptoAlgorithm.ALG_AES_256_GCM_IV12_TAG16_NO_KDF;
+      case ALG_AES_128_GCM_IV12_TAG16_HKDF_SHA256:
+        return CryptoAlgorithm.ALG_AES_128_GCM_IV12_TAG16_HKDF_SHA256;
+      case ALG_AES_192_GCM_IV12_TAG16_HKDF_SHA256:
+        return CryptoAlgorithm.ALG_AES_192_GCM_IV12_TAG16_HKDF_SHA256;
+      case ALG_AES_256_GCM_IV12_TAG16_HKDF_SHA256:
+        return CryptoAlgorithm.ALG_AES_256_GCM_IV12_TAG16_HKDF_SHA256;
+      case ALG_AES_128_GCM_IV12_TAG16_HKDF_SHA256_ECDSA_P256:
+        return CryptoAlgorithm.ALG_AES_128_GCM_IV12_TAG16_HKDF_SHA256_ECDSA_P256;
+      case ALG_AES_192_GCM_IV12_TAG16_HKDF_SHA384_ECDSA_P384:
+        return CryptoAlgorithm.ALG_AES_192_GCM_IV12_TAG16_HKDF_SHA384_ECDSA_P384;
+      case ALG_AES_256_GCM_IV12_TAG16_HKDF_SHA384_ECDSA_P384:
+        return CryptoAlgorithm.ALG_AES_256_GCM_IV12_TAG16_HKDF_SHA384_ECDSA_P384;
+      case ALG_AES_256_GCM_HKDF_SHA512_COMMIT_KEY:
+        return CryptoAlgorithm.ALG_AES_256_GCM_HKDF_SHA512_COMMIT_KEY;
+      case ALG_AES_256_GCM_HKDF_SHA512_COMMIT_KEY_ECDSA_P384:
+        return CryptoAlgorithm.ALG_AES_256_GCM_HKDF_SHA512_COMMIT_KEY_ECDSA_P384;
+      default:
+        throw new IllegalArgumentException("Unrecognized ESDK algorithmSuiteId: " + esdkAlgorithmSuiteId);
+    }
+  }
+
   public interface Builder {
-    Builder impl(ESDK impl);
+    Builder impl(AwsCrypto impl);
 
-    ESDK impl();
+    AwsCrypto impl();
 
     TestESDK build();
   }
 
   static class BuilderImpl implements Builder {
 
-    protected ESDK impl;
+    protected AwsCrypto impl;
 
     protected BuilderImpl() {}
 
-    public Builder impl(ESDK impl) {
+    public Builder impl(AwsCrypto impl) {
       this.impl = impl;
       return this;
     }
 
-    public ESDK impl() {
+    public AwsCrypto impl() {
       return this.impl;
     }
 
